src/rust/engine/src/lib.rs

```diff
@@ -739,14 +739,15 @@ pub extern "C" fn capture_snapshots(
 
   with_scheduler(scheduler_ptr, |scheduler| {
     let core = scheduler.core.clone();
-    core.block_on(
+    core.executor.block_on(
```

Comment:
If the tasks inside the `join_all` all have independent Tasks associated with them, is it necessary for the join to use `block_on`? Feels like not, because none of them should cause the Task running the join to "suspend".

This also feels like it applies to the other methods that are using `block_on`: if each IO operation is spawning a new task (similar to "spawning a new thread"), then none of them are causing their parent task to suspend.